### PR TITLE
feat(providers): add Duende IdentityServer 6

### LINKED ISSUE #4830: IdentityServer4 is now IdentityServer Duende
### What is the improvement or update you wish to see?

Change the name or add another provider with that name and fix all the links in the documentation of the provider. 

For example, the demo now is: https://demo.duendesoftware.com/

### Is there any context that might help us understand?

Duende have taken over maintenance of the IS4 libraries

### Does the docs page already exist? Please link to it.

https://next-auth.js.org/providers/identity-server4

### PR DESCRIPTION
**Pull request by josh-the-dev**
## ☕️ Reasoning

What changes are being made? What feature/bug is being fixed here?

Next-auth only supported Identity Server 4. This PR is adding support for Identity Server 6 found : [Identity Server 6](https://duendesoftware.com/])

It's effectively a copy of the work done for Identity Server 4, but added as a new provider, as that should be sufficient for this to work.

Given people are still using Identity Server 4 as well, I'd rather not replace that provider, so I added the Identity Server 6 provider as a new one.

_As a side note, this is my first PR here, so if there's anything I have done incorrectly please give me feedback 👍_

## 🧢 Checklist

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

## 🎫 Affected issues

Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR.

Fixes:  #4830 

## 📌 Resources

- [Contributing guidelines](./CONTRIBUTING.md)
- [Code of conduct](./CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/next-auth/src/providers/duende-identity-server6.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { OAuthConfig, OAuthUserConfig } from "./oauth"
```
Comment (balazsorban44) [inline packages/next-auth/src/providers/duende-identity-server6.ts:1]:
```suggestion
import type { OAuthConfig, OAuthUserConfig } from "./oauth"
```

File: packages/next-auth/src/providers/duende-identity-server6.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+export interface DuendeISUser extends Record<string, any> {
+  email: string
+  id: string
+  name: string
+  verified: boolean
+}
+
+/** @type {import(".").OAuthProvider} */
```
Comment (balazsorban44) [inline packages/next-auth/src/providers/duende-identity-server6.ts:10]:
```suggestion
```

Unnecessary in TS files.

File: docs/versioned_docs/version-v3/providers/duende-identity-server6.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  Providers.DuendeIdentityServer6({
+    id: "demo-identity-server",
+    name: "Demo IdentityServer6",
+    scope: "openid profile email api offline_access",
+    domain:  "demo.duendesoftware.com/",
+    clientId: "interactive.confidential",
+    clientSecret: "secret",
+    protection: "pkce"
+  })
+]
```
Comment (balazsorban44) [inline docs/versioned_docs/version-v3/providers/duende-identity-server6.md:57]:
```suggestion
import DuendeIDS6Provider from "next-auth/providers/duende-identity-server6"
...
providers: [
  DuendeIDS6Provider({
    clientId: process.env.DUENDE_IDS6_ID,
    clientSecret: process.env.DUENDE_IDS6_SECRET,
    issuer: "demo.duendesoftware.com",
  })
]
```

File: docs/versioned_docs/version-v3/providers/duende-identity-server6.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+---
+id: duende-identityserver6
+title: DuendeIdentityServer6
+---
+
+## Documentation
+
+https://duende-identityserver6.readthedocs.io/en/latest/
```
[the remark this answers is not in the document]
Comment (josh-the-dev) [reply docs/versioned_docs/version-v3/providers/duende-identity-server6.md:8]: @balazsorban44  how have we done provider docs in the past? Sorry I've not tackled that before. I mainly looked at the identity server 4 implementation to try and in essence recreate that.

File: packages/next-auth/src/providers/duende-identity-server6.ts (hunk 2 above)
Comment (josh-the-dev) [reply packages/next-auth/src/providers/duende-identity-server6.ts:10]: Oh this is brilliant to know, thanks

File: docs/versioned_docs/version-v3/providers/duende-identity-server6.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (balazsorban44) [reply docs/versioned_docs/version-v3/providers/duende-identity-server6.md:8]: Just googled it. :slightly_smiling_face: Here is the link 
```suggestion
https://docs.duendesoftware.com/identityserver/v6
```

File: apps/dev/pages/api/auth/[...nextauth].ts [hunk 5]
```diff
@@ -150,6 +151,11 @@ export const authOptions: NextAuthOptions = {
       clientSecret: process.env.IDS4_SECRET,
       issuer: process.env.IDS4_ISSUER,
     }),
+    DuendeIDS6Provider({
+      clientId: process.env.DUENDE_IDS6_ID,
+      clientSecret: process.env.DUENDE_IDS6_SECRET,
+      issuer: process.env.DUENDE_IDS6_ISSUER,
+    }),
```
Comment (balazsorban44) [inline apps/dev/pages/api/auth/[...nextauth].ts:158]:
```suggestion
    DuendeIDS6Provider({
      clientId: "interactive.confidential",
      clientSecret: "secret",
      issuer: "https://demo.duendesoftware.com",
    }),
```

File: docs/versioned_docs/version-v3/providers/duende-identity-server6.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+```js
+import DuendeIDS6Provider from "next-auth/providers/duende-identity-server6"
+...
+providers: [
+  DuendeIDS6Provider({
+    clientId: process.env.DUENDE_IDS6_ID,
+    clientSecret: process.env.DUENDE_IDS6_SECRET,
+    issuer: "demo.duendesoftware.com",
+  })
```
Comment (balazsorban44) [inline docs/versioned_docs/version-v3/providers/duende-identity-server6.md:50]:
```suggestion
  DuendeIDS6Provider({
    clientId: "interactive.confidential",
    clientSecret: "secret",
    issuer: "https://demo.duendesoftware.com",
  })
```

File: apps/dev/pages/api/auth/[...nextauth].ts (hunk 5 above)
Comment (josh-the-dev) [reply apps/dev/pages/api/auth/[...nextauth].ts:158]: Would we not want this as environment variables? Interested in the choice to hardcode them this way instead!

File: docs/versioned_docs/version-v3/providers/duende-identity-server6.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (josh-the-dev) [reply docs/versioned_docs/version-v3/providers/duende-identity-server6.md:8]: Ah brilliant, my mistakes I thought these were links to docs hosted by next-auth. Not a problem will commit this suggestion.

File: docs/versioned_docs/version-v3/providers/duende-identity-server6.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+## Documentation
+
+https://docs.duendesoftware.com/identityserver/v6
+
+## Options
+
+The **DuendeIdentityServer6 Provider** comes with a set of default options:
+
+- [DuendeIdentityServer6 Provider options](https://github.com/nextauthjs/next-auth/blob/main/src/providers/duende-identity-server6.ts)
```
Comment (balazsorban44) [inline docs/versioned_docs/version-v3/providers/duende-identity-server6.md:14]:
```suggestion
- [DuendeIdentityServer6 Provider options](https://github.com/nextauthjs/next-auth/tree/main/packages/next-auth/src/providers/duende-identity-server6.ts)
```

File: apps/dev/pages/api/auth/[...nextauth].ts (hunk 5 above)
Comment (josh-the-dev) [reply apps/dev/pages/api/auth/[...nextauth].ts:158]: Alright cool yeah that's right given its only for demo purposes that seems fine to me :)

**Review by balazsorban44 — APPROVED** [review]
Thanks!

**Pull request merged**